olive/auto_optimizer/regulate_mixins.py

```diff
@@ -0,0 +1,124 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+# --------------------------------------------------------------------------
+
+from copy import deepcopy
+
+from olive.hardware.accelerator import Device
+
+
+class RegulatePassConfigMixin:
+    def regulate_pass_flows_dict(self, pass_flows_dict):
+        # special passes: OrtTransformerOptimization and OrtPerfTuning can be used for both fp16 and fp32
+        # we need assign different pass name for them
+        # for example: gpu_cuda_fp16, we need rename OrtTransformerOptimization to OrtTransformerOptimization_cuda_fp16
+        pass_flows_by_fp16 = pass_flows_dict.get("fp16", [])
+        pass_config, pass_flows_16 = self.regulate_fp16(None, pass_flows_by_fp16)
+
+        # flatten pass_flows_dict to pass_flows and generate the default pass_configs
+        pass_flows = []
+        unique_pass_flows = set()
+        if pass_flows_16:
+            pass_flows_dict["fp16"] = pass_flows_16
+        for pfs in pass_flows_dict.values():
+            for pf in pfs:
+                if tuple(pf) not in unique_pass_flows:
+                    pass_flows.append(pf)
+                unique_pass_flows.add(tuple(pf))
+                for p in pf:
+                    if p not in pass_config:
+                        pass_config.update({p: {"type": p, "config": {}}})
+
+        return pass_config, pass_flows
+
+    def regulate_fp16(self, pass_config, pass_flows):
```

Comment:
>regulate_fp16 [](http://example.com/codeflow?start=8&length=13)

Basically, it a method is only invoked in the class itself, we can safely make it private. ditto for allow_precision

Reply:
Set as private in latest commit.